crates/web5/src/credentials/verifiable_credential_1_1.rs

```diff
@@ -515,18 +515,9 @@ impl JosekitJwsVerifier for JoseVerifier {
     }
 
     fn verify(&self, message: &[u8], signature: &[u8]) -> core::result::Result<(), JosekitError> {
-        let result = self
-            .verifier
+        self.verifier
             .verify(message, signature)
-            .map_err(|e| JosekitError::InvalidSignature(e.into()))?;
-
-        match result {
-            true => Ok(()),
-            false => Err(JosekitError::InvalidSignature(
-                // 🚧 improve error message semantics
-                DsaError::VerificationFailure("ed25519 verification failed".to_string()).into(),
-            )),
-        }
+            .map_err(|e| JosekitError::InvalidSignature(e.into()))
```

Comment:
nice, much simpler.

Reply:
yeah I think so too, I'm sold on the monad pattern